execute/README.md

```diff
@@ -0,0 +1,456 @@
+Some transformations, like `map()`, will always output the same number of rows they receive in a chunk.
+These are the easiest.
+Others, like `filter()`, might reduce the length of the array and should determine the new length of the table chunk in advance.
+There are also cases where a transformation might need to rearrange data from different buffers or could produce more data.
+For these circumstances, chunk sizes should be limited to [table.BufferSize](https://pkg.go.dev/github.com/influxdata/flux/execute/table#pkg-constants).
+
+**It is not required that a transformation determine the length of a table chunk before producing one**, but it is highly advised.
+Memory reallocation during table chunk creation is a top contributor to slowdown.
+
+### Construct the array for each column
+
+We produce an array for each column in the table chunk using the [github.com/influxdata/flux/array](https://pkg.go.dev/github.com/influxdata/flux/array) package.
+Each flux type corresponds to an array type according to the following table:
+
+|Flux Type|Arrow Type
+|---------|----------
+|Int|Int
+|UInt|Uint
+|Float|Float
+|String|String
+|Bool|Boolean
+|Time|Int
+
+At its simplest, creating an array is done using the given skeleton.
+
+    b := array.NewIntBuilder(mem)
+    b.Resize(10)
+    for i := 0; i < 10; i++ {
+        b.Append(int64(i))
+    }
+    return b.NewArray()
+
+Other techniques for building arrays efficiently are contained in the arrow arrays section below.
+
+### Construct and send a table.Chunk using arrow.TableBuffer
+
+We construct a [table.Chunk](https://pkg.go.dev/github.com/influxdata/flux/execute/table#Chunk) using [arrow.TableBuffer](https://pkg.go.dev/github.com/influxdata/flux/arrow#TableBuffer).
+
+    buffer := arrow.TableBuffer{
+        GroupKey: execute.NewGroupKey(...),
+        Columns: []flux.ColMeta{...},
+        Values: []array.Interface{...},
+    }
+    chunk := table.ChunkFromBuffer(buffer)
+
+    if err := d.Process(chunk); err != nil {
+        return err
+    }
+
+## Arrow Arrays
+
+When constructing arrow arrays, there are some general guidelines that apply to every transformation.
+
+### Preallocate Memory
+
+Preallocate memory by determining the size of a chunk in advance and using `Resize()` to set the capacity of the builder.
+For strings, it is also helpful to preallocate memory for the data using `ReserveData` if this can be easily known.
+String appends are usually the biggest performance sink for efficiency.
+
+### Limit Chunk Sizes
+
+Limit chunk sizes to [table.BufferSize](https://pkg.go.dev/github.com/influxdata/flux/execute/table#pkg-constants).
+The array values in a column are contained in contiguous data.
+When the chunk size gets larger, the memory allocator has to find a spot in memory that fits that large size.
+Larger chunk sizes are generally better for performance, but too large of buffer sizes put too much pressure on the memory allocator and garbage collector.
```

Comment:
nit/ Maybe: `...but buffer sizes that are too large put too much pressure on the ...`